README.md

```diff
@@ -100,71 +100,19 @@ $ pnpm add telegraf
 
 ### `Telegraf` class
 
-[`Telegraf`](https://telegraf.js.org/classes/telegraf.html) instance represents your bot. It's responsible for obtaining updates and passing them to your handlers.
+[`Telegraf`] instance represents your bot. It's responsible for obtaining updates and passing them to your handlers.
 
 ### `Context` class
 
-`Telegraf` creates one [`Context`](https://telegraf.js.org/classes/context.html) for each incoming update.
+`ctx` you can see in every example is a [`Context`] instance.
+[`Telegraf`] creates one for each incoming update and passes it to your middleware.
 It contains the `update`, `botInfo`, and `telegram` for making arbitrary Bot API requests,
 as well as shorthand methods and getters.
 
 This is probably the class you'll be using the most.
 
-### Middleware
-
-Middleware is an essential part of any modern framework.
```

Comment:
@wojpawlik Why did you delete this good explanation?

Reply:
Because I wrote a [better one (IMO) below](https://github.com/telegraf/telegraf/blob/readme/README.md#middleware).

Reply:
And why is it better? As a beginner i didn't understand anything in your explanation

Reply:
We have a poll which explanation is better: https://t.me/TelegrafJSChat/18314

Personally, I found the old section very roundabout. It has whooping 9 paragraphs, some of them have issues:

> It allows you to modify requests and responses as they pass between the Telegram and your bot.

Very sloppy. It doesn't mention that it's possible to handle or swallow updates as well. Modifying responses is messy until #1267.

> You can imagine middleware as a chain of logic connection your bot to the Telegram request.

I don't understand this at all.

> **`ctx`** is the context for one [Telegram update](https://core.telegram.org/bots/api#update).

A separate section for `Context` says this already.

> **`next`** is a function that is invoked to execute the downstream middleware.
> It returns a `Promise` with a function `then` for running code after completion.

Correct, but should mention `await`.

So, what exactly is missing from the new explanation?

Reply:
![image](https://user-images.githubusercontent.com/4659199/105244076-a5c0a100-5b78-11eb-8dc5-327cba8aae12.png)
IDK, it feels a bit missing smth, like lack of explanation what is it for, just and explanation of how to use next callback

![image](https://user-images.githubusercontent.com/4659199/105244094-ae18dc00-5b78-11eb-89f6-eee99e5a7464.png)
In the old version there is a good explanation about a chain of logic which IMO should be left here for some people who are less experienced with modern JS frameworks and have almost no experience with middleware

Like newcomers to Node 
or
Frontenders who use mostly smth like Vue
or
Backenders who have moderate experience with smth like NestJS and didn't come across writing own middlewares for it

Not all people use vanilla express, you know

And yes, the part about modifying req and res is essential too

Reply:
> > It allows you to modify requests and responses as they pass between the Telegram and your bot.
> 
> Very sloppy. It doesn't mention that it's possible to handle or swallow updates as well. Modifying responses is messy until #1267.

1. Then maybe it should be mentioned? About updates. Why delete partial explanation if you can improve it by adding what is missing?

2. If it is messy in any way then maybe it should be mentioned? Like that there is a possibility of modifying the response, but be cautious if you do such thing and such thing and such thing

Reply:
>> You can imagine middleware as a chain of logic connection your bot to the Telegram request.

> I don't understand this at all.

Seems like there was some poor translation to English which might need rewriting, not deletion.
Need to think how to rewrite.